### DISCUSSION TITLE
Unterschied  opendtu-generic.bin vs  opendtu-generic_esp32.bin

### DISCUSSION THREAD

**Discussion by Kenaschon** [tbnobody/OpenDTU#1509] [General]
Hallo zusammen,

wie in der Überschrift schon geschrieben.

Worin liegt der Unterschied zwischen den beiden Dateien?

Ich habe eine NMCU-ESP32. Opendtu zeigt das an - ESP32-D0WDQ6-V3

Sollte ich dann besser diese Firmware nehmen? opendtu-generic_esp32s3.bin 

Danke für dieses tolle Projekt.

**Comment by Tiese** [comment]
Hallo @Kenaschon,

da du **nicht** den **_ESP32-S3_** hast, solltest du die **_opendtu-generic_esp32s3.bin_** auch **nicht** nehmen!

Schöne Grüße
Tiese

**Comment by Kenaschon** [comment]
Ich dacht das V3 gleich S3 ist.

Danke für die Info.

**Reply by Andi161** [reply]
Modell-Name decoded:
ESP32-D0WDQ6-V3
D = Dual Core
0 = no in-package flash included
WD = Wifi + Bluetooth Dual mode
Q6 = package, 6*6

Wenn statt D ein S steht, bedeutet das Single Core. Die 3 bei S3 bedeutet, dass der Chip einen eingebauten Flash Speicher hat, während deine DualCore Versionen einen externen Flash-Speicher haben. Deswegen D0.

**Comment by Forstgrund** [comment]
Ja da könnte mal ein kleiner Hinweis zu den Downloaddateien stehen....ich bin mir auch unsicher....
Ich habe den ESP32-D0WD-V3 und lade immer nur opendtu-generic.bin
Ob ich die  opendtu-generic_esp32.bin besser nehmen soll?--->Test: zeigt keine Livedaten
Gibt es irgendo eine Erläuterung die Licht ins Dunkel bringt?

**Reply by Andi161** [reply]
Stimme dir voll zu. Hab heute den ganzen Tag herumgetüftelt um herauszufinden, warum der NRF24 sich nicht mit dem Hoymiles verbinden will. Erst über diesen Thread ist es mir klar geworden.
Mit dem Begriff Device-File konnte ich nichts anfangen. Und ein Pin-Mapping file konnte ich ja auch nicht sichern, da bei der alten Firmware die Option gar nicht angeboten wurde. In sofern war es nicht schlüssig, wieso ich bei der neuen Firmware ein entsprechendes File hochladen muss und das NRF24 Profil aktivieren soll.
Ich habe ja nichts dagegen, etwas dazu zu lernen. Aber wenn das neue Wissen nur einmal im Leben benötigt wird, da bei den nächsten Updates das Profil bereits im Speicher vorhanden ist, ist es nicht so prickelnd, dafür soviel Zeit zu investieren.

Für alle anderen Leser, die wie ich schon bei den Grundlagen des Flashens Schwierigkeiten haben:
1. Den ESP mit dem USB Port des Computers zu verbinden reicht nicht. Ihr müsst auch noch einen Virtuellen ComPort Treiber für den UART auf der Platine installieren. Erst dann funktioniert das Web-Flash Tool. In meinem Fall war es der FTDI VCP-Treiber. Anschließend zeigt das Web-Flash-Gui auch den richtigen Port an. Nur kannst du dann nicht auswählen, welches bin-File du hochlädst. Das scheint automatisch die neueste Version zu laden, ob Factory-Version oder nicht- no idea. Ich habe dann nach dem Flashen einen OTA-Update mit der openDTU-generic.bin gemacht. Damit funktionierte die Verbindung zum HM wieder und ich wusste, dass das Teil doch funktioniert. Also nochmal geflashed und mit Punkt 2 weitergemacht.
2. Wenn dann erfolgreich geflashed wurde und ihr das vorher gesicherte config.json File hochladet, müsst ihr die falsche Seriennummer im DTU Menu korrigieren. Und natürlich das richtige Device-File laden (bei mir war es das nodemco_esp32.json). Und dann im Hardware-Menü die Konfiguration NRF24 auswählen und speichern. Dass das Device-File im GUI als Pin-Mapping-File angesprochen wird, werdet ihr dann auch verschmerzen. Weil - jetzt funktioniert es.

**Comment by NintendoTito** [comment]
Hatte die Frage auch. Hier stehen noch mehr Infos:

https://github.com/tbnobody/OpenDTU/discussions/1619

**Comment by tbnobody** [comment]
Hmmm Doku gucken? https://tbnobody.github.io/OpenDTU-docs/firmware/flash_esp/
Bzw. was würdest du hier vermissen? Dann ergänze ich das.

**Reply by ghost** [reply]
Dem schließe ich mich voll an.
Heute in der App-verwöhnten Welt liest kaum einer mehr was. Ich muss mich auch zwingen.

Ich errichte "Gefahrenmeldeanlagen" (neudeutsch). Zu meinen Kollegen, die mich wegen jeder Kleinigkeit anrufen, sage ich immer:
Eine Installationsanleitung ist nicht nur Füllmaterial in der Verpackung.

@tbnobody 
Man muss immer wissen, was für einen ESP32 man hat, auch beim Web Flasher ganz unten.
Der [Community Webflasher](https://solar.metacontrol.eu/opendtu_webinstall/) (Danke @MetaChuh) ist für ganz Unbetuchte da besser.
Vielleicht könnte der bei der Flashanleitung mit verlinkt werden.

**Reply by ghost** [reply]
@tbnobody
Bzw. wäre Hilfe bei der Doku genehm? Ich habe da schon mal eine kleine Änderung angestoßen die auch umgesetzt wurde.
Ich bin eher Laie als Fachmann und stoße in der Doku dadurch eher mal auf Fragen als ein Entwickler.
Ich traue mich da nicht immer ein Pullrequest anzustoßen, weil ich denke, das nervt euch.

**Reply by tbnobody** [reply]
> Bzw. wäre Hilfe bei der Doku genehm?

Ja, gerne PR's bauen.

**Comment by kptkip** [comment]
> Hmmm Doku gucken? https://tbnobody.github.io/OpenDTU-docs/firmware/flash_esp/ Bzw. was würdest du hier vermissen? Dann ergänze ich das.

Was mir auf dieser Seite fehlt, ist eine Erläuterung warum es überhaupt unterschiedliche FW-Pakete gibt.

Natürlich ist es mit etwas Durchlesen erkennbar (wenn auch nicht sehr explizit), dass es unterschiedliche Pakete für:
1. ESP32-MCUs und ESP32-S3-MCUs 
2. welche für das OTA und First Flash gibt. Soweit so gut.

Allerdings ist nicht erkennbar was die Unterschiede zwischen
`opendtu-generic.bin` und `opendtu-generic_esp32.bin`
sind, oder
`opendtu-generic.factory.bin` und `opendtu-generic_esp32.factory.bin`

Wenn ich es richtig verstanden habe, muss ich im Falle der `*_esp32.bin` noch _Device Profiles_ hinzufügen. Was das aber konkret sind ( https://www.opendtu.solar/firmware/device_profiles/#implemented-configuration-values ist alles andere als selbsterklärend) und warum da überhaupt ein Unterschied existiert, wird nicht erläutert.

Das einzige, was ich verstanden habe ist, dass die "generic.bin" _deprecated_ sind. Da es aber die einzige Version ist, die bei mir out-of-the-box tut, bleibt sie auch erstmal drauf.
